.circleci/config.yml

```diff
@@ -1,29 +1,32 @@
-version: 2
-jobs:
-  test:
-    machine: true
-    working_directory: /home/circleci/go/src/github.com/geckoboard/sql-dataset
-    environment:
-      GOPATH: "/home/circleci/go"
+version: 2.1
+
+machine: &machine_setup
+  machine:
+    image: ubuntu-2004:202010-01
+  working_directory: /home/circleci/go/src/github.com/geckoboard/sql-dataset
+  environment:
+    GOPATH: "/home/circleci/go"
```

Comment:
I'm a bit confused as to how this is working - I didn't think you could specify `machine:` at a top level outside of a job...? Regardless, rather than rely on the `&` yaml syntax it might be better to use [circleci's executor section](https://circleci.com/docs/2.0/configuration-reference/#executors-requires-version-21) to pull out the configuration of the environment - it's a little less magical than `<<: *`

Here's an example of using it: https://github.com/geckoboard/juno/blob/master/.circleci/config.yml#L58

Reply:
NIce 👍  agreed